artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonAbstractReceiver.java

```diff
@@ -91,8 +92,8 @@ public AMQPSessionContext getSessionContext() {
       return protonSession;
    }
 
-   protected void recoverContext() {
-      sessionSPI.recoverContext();
+   protected OperationContext recoverContext() {
```

Comment:
Think this could use the matching javadoc to describe that it is really the _current/old_ context being returned, not the 'recovered' one that will be used afterwards.